src/main/java/nextstep/courses/infrastructure/JdbcSessionRepository.java

```diff
@@ -0,0 +1,38 @@
+package nextstep.courses.infrastructure;
+
+import nextstep.courses.domain.session.Session;
+import nextstep.courses.domain.session.SessionRepository;
+import nextstep.courses.domain.session.SessionStatus;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Repository("sessionRepository")
+public class JdbcSessionRepository implements SessionRepository {
+    private JdbcOperations jdbcTemplate;
+
+    public JdbcSessionRepository(JdbcOperations jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public Session findById(Long id) {
+        String sql = "select id, status, start_date_time, end_date_time, free, max_attendance from session where id = ?";
+        RowMapper<Session> rowMapper = (rs, rowNum) -> new Session(
+                rs.getLong(1),
+                SessionStatus.valueOf(rs.getString(2)),
+                rs.getTimestamp(3).toLocalDateTime(),
+                rs.getTimestamp(4).toLocalDateTime(),
+                rs.getBoolean(5),
+                rs.getInt(6));
+        return jdbcTemplate.queryForObject(sql, rowMapper, id);
+    }
+
+    @Override
+    public void save(Session session) {
+        String sql = "insert into session (course_id, cover_image_id, status, start_date_time, end_date_time, free, max_attendance) values (?, ?, ?, ?, ?)";
+        jdbcTemplate.update(sql, session.getCourse().getId(), session.getCoverImg().getId(),
+                session.getSessionStatus().name(), session.getPeriod().getStartDateTime(),
```

Comment:
```suggestion
                session.name(), session.startDate(),
```

> ### 객체지향 생활체조 원칙
- 규칙 4: 한 줄에 점을 하나만 찍는다.

캡슐화를 진행하였고, 외부에서 객체의 내부 상황을 모르게 하는게 더 좋지 않을까요?

`session.name` 이름이 궁금
`session.startDate` 시작일이 궁금

외부에서는 궁금한것이 무엇인지 단순화 시켜보시면 좋을것 같아요~

[[OOP] 디미터의 법칙(Law of Demeter)](https://mangkyu.tistory.com/147)

Reply:
피드백 감사합니다!!

레퍼지토리 부분이라 괜찮지 않을까 라는 생각을 좀 가지고 있었는데, 최대한 객체 지향적으로 코드를 작성하도록 좀 더 노력해보겠습니다

감사합니다